src/Mutant.php

```diff
@@ -48,6 +49,11 @@ class Mutant
      */
     protected $process;
 
+    /**
+     * @var MutantResult
+     */
+    protected $result;
```

Comment:
Hum... I may be mistaken on the responsibilities of the classes, but Mutant seems to me it is more than just a data container of the applied mutation, considering it also holds the $process property, and has the toArray() method, which returns among other thing the result of the process (stdout and stderr). 

I'll be pushing another commit to remove the dependency on $tracker and $group, which resolves part of the issue.